src/build/rewrite-webcomponents-loader.ts

```diff
@@ -12,16 +13,43 @@ const attrValueMatches = (attrName: string, regex: RegExp) => {
   };
 };
 
-const p = dom5.predicates;
-const scriptIncludeWebcomponentsLoader = p.AND(
-    p.hasTagName('script'),
-    attrValueMatches('src', /\bwebcomponents-loader\.js$/));
+const webcomponentsLoaderMatcher = predicates.AND(
+    predicates.hasTagName('script'),
+    attrValueMatches('src', /\bwebcomponents\-(loader|lite)\.js$/));
+const headMatcher = predicates.hasTagName('head');
+const bodyMatcher = predicates.hasTagName('body');
+const scriptMatcher = predicates.AND(
+    predicates.hasTagName('script'),
+    predicates.OR(
+        webcomponentsLoaderMatcher,
+        predicates.NOT(predicates.hasAttr('type')),
+        predicates.hasAttrValue('type', 'text/javascript'),
+        predicates.hasAttrValue('type', 'application/javascript')));
+const linkMatcher = predicates.AND(
+    predicates.hasTagName('link'),
+    predicates.OR(
+        predicates.hasAttrValue('rel', 'import'),
+        predicates.hasAttrValue('rel', 'stylesheet')));
+const styleMatcher = predicates.AND(
+    predicates.hasTagName('style'),
+    predicates.OR(
+        predicates.NOT(predicates.hasAttr('type')),
+        predicates.hasAttrValue('type', 'text/css')));
+
 
 /**
  * When compiling ES6 classes down to ES5 we need to include a special form of
- * the webcomponents loader to be compatible with native custom elements.
+ * the webcomponents loader to be compatible with native custom elements. This
+ * includes moving the loader script element and all relevent siblings
+ * following it into the body so that the <ES5 polyfill can be properly
```

Comment:
Instead of ES5 polyfill can we be more specific and say it is the Custom Elements ES5 adapter?